src/DI/ServiceLookup/ServiceProviderEngine.cs

```diff
@@ -57,16 +86,11 @@ protected ServiceProviderEngine(IEnumerable<ServiceDescriptor> serviceDescriptor
             return _ => null;
         }
 
-        internal object GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Did you compare perf inlined vs non-inlined?

Reply:
No, but we usually extract throwing into non inlined methods so their callers are not increasing in size and inline easier.